### ISSUE TITLE
[AutoDiscovery] Develop auto-discovery for OctoPrint printers

### ISSUE THREAD

**Issue by davidzwa** [fdm-monster/fdm-monster#1686]
Implement a method to auto-discover printers.

## Possible solutions
- [ ] MQTT
- [ ] SSDP
- [ ] ZeroConf

## Possible installations
- [ ] OctoPrint
- [ ] Moonraker (Fluidd / Mainsail)

**Comment by davidzwa** [comment]
Currently leaning towards installing https://plugins.octoprint.org/plugins/mqtt/ for autodiscovery

**Comment by davidzwa** [comment]
Not a good sign https://github.com/OctoPrint/OctoPrint-MQTT/issues/131

**Comment by davidzwa** [comment]
Requires HA but drops the need for Mosquitto https://plugins.octoprint.org/plugins/homeassistant/
Requires Mosquitto but is easy to break https://plugins.octoprint.org/plugins/mqttpublish/
Bridges the API to MQTT https://github.com/jneilliii/OctoPrint-MQTTSubscribe

**Comment by davidzwa** [comment]
https://cedalo.com/blog/nodejs-mqtt-usage-with-mosquitto-example/
https://iot-blogger.com/mqtt-node-js/

**Comment by mkevenaar** [comment]
Help wanted
I have implemented SSDP before, but I experienced issues where it was slow or inconsistent (mostly docker). Another doubt I had with it, when I was working on this for OctoFarm, was that I did not know how to complete the authentication triangle (asymmetric crypto / key sharing).

Why SSDP
It's a protocol that is supported by most networks/routers and which does not require separate brokers or applications (compared to MQTT). Furthermore, OctoPrint itself already has it built in, where as MQTT would require two plugins or a newly written plugin with the downside of having to depend on a plugin (instead of a built-in/pre-packaged feature). This does not mean I do not see the benefits of MQTT and a dedicated broker, it's just that I realize SSDP is a nice method to discover OctoPrints out of the box.
